### PR TITLE
Add ability to use Activated BatchNorm in decoder.

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
Hi! This changeset adds ability to use InPlace Activated Batch Normalization(https://arxiv.org/abs/1712.02616) in decoder which must decrease memory consumption. I propose to not add any ABN implementation as a dependency, because from my point of view existing implementations are not stable enough, but we allow user to provide his own implementation(or some of existing ones).

### CODE REVIEW

**Comment by qubvel** [comment]
Hi, not sure it is a good idea to provide class object to model constructor
Any other ideas?

**Comment by SiarheiFedartsou** [comment]
> Hi, not sure it is a good idea to provide class object to model constructor
> Any other ideas?

Hi! You can provide something like "factory function" instead of class object(see tests). What's wrong with such solution? As an alternative we can create InPlaceABN directly in Conv2DReLU, but in such case we must add some implementation of InPlaceABN as a dependency and it is not so flexible.
Or maybe the issue is naming? Should we rename `abn_class` to something like `abn_factory`?

**Comment by qubvel** [comment]
I mean it is not possible to make e.g. yaml or json config to initialize model
So, my thoughts to make some optinal dependencies and raise an error if Inplace BN is not installed

**Comment by SiarheiFedartsou** [comment]
Okay, good point, it was in my first implementation of this stuff :) Will make it when will have some time for it(think, during a couple of days). Thanks!

**Comment by qubvel** [comment]
One more comment here, lets combine this option with 
`decoder_use_batchnorm`
 - False
 - True
 - "inplace"

**Comment by SiarheiFedartsou** [comment]
Updated.
